lib/ProductOpener/Ingredients.pm

```diff
@@ -3368,23 +3368,52 @@ sub get_missing_ecobalyse_ids ($ingredients_ref) {
 		delete $ingredient_ref->{ecobalyse_code};
 		delete $ingredient_ref->{ecobalyse_proxy_code};
 
-		# Getting properties from product
-		#my $product_origin = has_specific_ingredient_property()
-
-		# Getting the correct ecobalyse comake devde
-		my $ecobalyse_code = get_inherited_property("ingredients", $ingredient_ref->{id}, "ecobalyse:en");
-		if (defined $ecobalyse_code) {
-			$ingredient_ref->{ecobalyse_code} = $ecobalyse_code;
-		}
-		else {
-			my $ecobalyse_proxy_code
-				= get_inherited_property("ingredients", $ingredient_ref->{id}, "ecobalyse_proxy:en");
-			if (defined $ecobalyse_proxy_code) {
-				$ingredient_ref->{ecobalyse_proxy_code} = $ecobalyse_proxy_code;
-			}
-			else {
-				push(@ingredients_without_ecobalyse_ids, $ingredient_ref->{id});
-			}
+		# We are now looking for the appropriate ecobalyse id :
+    	# ecobalyse_origins_france_label_organic 			(if the product comes from france, and is organic)
+    	# ecobalyse_origins_european-union_label_organic	(if the product comes from europe, and is organic)
+    	# ecobalyse_label_organic							(if the product is organic)
+    	# ecobalyse_origins_france							(if the product comes from france)
+	    # ecobalyse_origins_european-uinion					(if the product comes from the Europe region)
+    	# ecobalyse 										(else)
+
+
+		# If the ingredient is organic...
+		if ($ingredient_ref->{labels} =~ /\ben:organic\b/) {
+
+		    # Retrieve the correct ecobalyse code
+		    my $ecobalyse_code = get_inherited_property("ingredients", $ingredient_ref->{id}, "ecobalyse_labels_en_organic:en");
```

Comment:
I think it's easier to build a list of  the possible suffixes, because even if we just want to support organic, we need to check those:

1. ecobalyse_labels_en_organic:en
2. ecobalyse_proxy_labels_en_organic:en
3. ecobalyse_en_organic:en
4. ecobalyse_proxy_en_organic:en

Here your current code checks 1. and 4.

You could add more if clauses to cover 2 and 3, but it's going to become difficult to read with a lot of duplicate code.

Instead just do something like this (pseudo code):

my @suffixes = ();
if (label organic) : push @suffixes, "_labels_organic"
push @suffixes, ''; # no suffix when we don't have anything

then a foreach loop on suffixes:

foreach my $suffix (@suffixes) {

foreach my $proxy ("", "_proxy") {

my $ecobalyse_code = get_inherited_property("ingredients", $ingredients_ref->{id}, "ecobalyse" . $proxy . $suffix)
if defined $ecobalyse_code {

..
last;
}
˚

Reply:
Indeed it's a much more better solution, i hadn't seen it like this : new pr incoming !